services/web/server/src/simcore_service_webserver/storage_api.py

```diff
@@ -28,16 +33,59 @@ def _get_storage_client(app: web.Application) -> Tuple[ClientSession, URL]:
     return session, endpoint
 
 
+async def get_storage_locations(
+    app: web.Application, user_id: UserID
+) -> FileLocationArray:
+    log.debug("getting %s accessible locations...", f"{user_id=}")
+    session, api_endpoint = _get_storage_client(app)
+    locations_url = (api_endpoint / "locations").with_query(user_id=user_id)
+    async with session.get(f"{locations_url}") as response:
+        response.raise_for_status()
+        locations_enveloped = Envelope[FileLocationArray].parse_obj(
+            await response.json()
+        )
+        assert locations_enveloped.data  # nosec
+        log.info("%s can access %s", f"{user_id=}", f"{locations_enveloped.data=}")
+        return locations_enveloped.data
+
+
+async def get_project_total_size(
+    app: web.Application, user_id: UserID, project_uuid: ProjectID
+) -> ByteSize:
+    log.debug("getting %s total size for %s", f"{project_uuid=}", f"{user_id=}")
+    user_accessible_locations = await get_storage_locations(app, user_id)
+    session, api_endpoint = _get_storage_client(app)
+
+    project_size_bytes = 0
+    for location in user_accessible_locations:
+        files_metadata_url = (
+            api_endpoint / "locations" / f"{location.id}" / "files" / "metadata"
+        ).with_query(user_id=user_id, uuid_filter=f"{project_uuid}")
+        async with session.get(f"{files_metadata_url}") as response:
+            response.raise_for_status()
+            list_of_files_enveloped = Envelope[list[FileMetaDataGet]].parse_obj(
+                await response.json()
+            )
+            assert list_of_files_enveloped.data  # nosec
+        for file_metadata in list_of_files_enveloped.data:
+            project_size_bytes += file_metadata.file_size
+    project_size = parse_obj_as(ByteSize, project_size_bytes)
+    log.info(
```

Comment:
Having a getter free function logging in INFO level the result is unnecessary. 
I think it is the decision of the caller to do that

Same applies for the other functions

Using the log_decorator would allow you to trace (in DEBUG) mode the inputs and outputs of the function. But that is something else

Reply:
I actually do not agree in this case, since this "getter" actually calls the storage micro-service.
I tend to go with:
- debug to enter the function
- info to leave the function